File: Strings/src/main/python/solution680.py
class Solution:
    def validPalindrome(self, s: str) -> bool:

        n = len(s)

        def skip_left():
            left = 0
            right = n - 1
            count = True

            while left < right:
                # print(left, right)
                if s[left] != s[right]:
                    if not count:
                        return False
                    left += 1
                    count = False
                    continue

                left += 1
                right -= 1

            return True

        def skip_right():
            left = 0
            right = n - 1
            count = True

            while left < right:
                # print(left, right)
                if s[left] != s[right]:
                    if not count:
                        return False
                    right -= 1
                    count = False
                    continue

                left += 1
                right -= 1

            return True

        print(skip_left(), skip_right())
        return skip_left() or skip_right()


solu = Solution()
s = "ebcbbececabbacecbbcbe"
#   "ebcbbcecabbacecebbcbe'
# s = "atbbga"
res = solu.validPalindrome(s)
print(res)

File: Strings/src/main/java/Solution680.java
public class Solution680 {
    public boolean validPalindrome(String s) {

        return check(s, 1);

    }

    private boolean check(String s, int times) {
        System.out.println(s);
        int left = 0;
        int right = s.length() - 1;

        while (left < right) {
            // System.out.println(left + " " + right);
            if (s.charAt(left) != s.charAt(right)) {
                if (times > 0) {
                    return check(s.substring(left, right), 0) || check(s.substring(left + 1, right + 1), 0);
                }
                return false;
            }
            left++;
            right--;
        }
        return true;
    }

    public static void main(String[] args) {
        Solution680 s = new Solution680();
        s.validPalindrome("abc");
        System.out.println(s.validPalindrome("abc"));
    }
}
